common/config.go

```diff
@@ -0,0 +1,240 @@
+}
+
+func WriteEdgeModulesYamlFile(path string) error {
+	modulesData := ModulesYaml{Modules: ModulesSt{Enabled: []string{"eventbus", "servicebus", "websocket", "metaManager", "edged", "twin", "dbTest"}}}
+	if err := Write2File(path, modulesData); err != nil {
+		return err
+	}
+	return nil
+}
+
+//MQTTMode = # 0: internal mqtt broker enable only. 1: internal and external mqtt broker enable. 2: external mqtt broker enable only.
+type MQTTMode uint8
+
+const (
+	MQTTInternalMode MQTTMode = iota
+	MQTTInternalExternalMode
+	MQTTExternalMode
+)
+
+//MQTTQoSType = # 0: QOSAtMostOnce, 1: QOSAtLeastOnce, 2: QOSExactlyOnce.
+type MQTTQoSType uint8
+
+const (
+	MQTTQoSAtMostOnce MQTTQoSType = iota
+	MQTTQoSAtLeastOnce
+	MQTTQoSExactlyOnce
+)
+
+type MQTTSt struct {
+	Server           string      `yaml:"server"`
+	InternalServer   string      `yaml:"internal-server"`
+	Mode             MQTTMode    `yaml:"mode"`
+	QOS              MQTTQoSType `yaml:"qos"`
+	Retain           bool        `yaml:"retain"`
+	SessionQueueSize uint64      `yaml:"session-queue-size"`
+}
+
+type WebSocketSt struct {
+	URL              string `yaml:"url"`
+	CertFile         string `yaml:"certfile"`
+	KeyFile          string `yaml:"keyfile"`
+	HandshakeTimeout uint16 `yaml:"handshake-timeout"`
+	WriteDeadline    uint16 `yaml:"write-deadline"`
+	ReadDeadline     uint16 `yaml:"read-deadline"`
+}
+
+type ControllerSt struct {
+	Placement           bool   `yaml:"placement"`
+	Heartbeat           uint32 `yaml:"heartbeat"`
+	RefreshAKSKInterval uint16 `yaml:"refresh-ak-sk-interval"`
+	AuthInfoFilesPath   string `yaml:"auth-info-files-path"`
+	PlacementURL        string `yaml:"placement-url"`
+	ProjectID           string `yaml:"project-id"`
+	NodeID              string `yaml:"node-id"`
+}
+
+type EdgeDSt struct {
+	RegisterNodeNamespace             string `yaml:"register-node-namespace"`
+	HostnameOverride                  string `yaml:"hostname-override"`
+	InterfaceName                     string `yaml:"interface-name"`
+	NodeStatusUpdateFrequency         uint16 `yaml:"node-status-update-frequency"`
+	DevicePluginEnabled               bool   `yaml:"device-plugin-enabled"`
+	GPUPluginEnabled                  bool   `yaml:"gpu-plugin-enabled"`
+	ImageGCHighThreshold              uint16 `yaml:"image-gc-high-threshold"`
+	ImageGCLowThreshold               uint16 `yaml:"image-gc-low-threshold"`
+	MaximumDeadContainersPerContainer uint16 `yaml:"maximum-dead-containers-per-container"`
+	DockerAddress                     string `yaml:"docker-address"`
+	Version                           string `yaml:"version"`
+}
+
+type EdgeHubSt struct {
+	WebSocket  WebSocketSt  `yaml:"websocket"`
+	Controller ControllerSt `yaml:"controller"`
+}
+
+type EdgeYamlSt struct {
+	MQTT    MQTTSt    `yaml:"mqtt"`
+	EdgeHub EdgeHubSt `yaml:"edgehub"`
+	EdgeD   EdgeDSt   `yaml:"edged"`
+}
+
+func WriteEdgeYamlFile(path string, modifiedEdgeYaml *EdgeYamlSt) error {
+
+	edgeID := "fb4ebb70-2783-42b8-b3ef-63e2fd6d242e"
```

Comment:
Why hard code edgeID here?

Reply:
This is just a default value. If you see the flow for edge.yaml's creation, this value will never be used, because we always have some valid modifiedEdgeYaml.EdgeHub.WebSocket.URL. It would have edgeID; either provided by user or generated. 
It is added just in case, if for some reason modifiedEdgeYaml is null or modifiedEdgeYaml.EdgeHub.WebSocket.URL is empty, we have to give some defaults to continue. Then user can himself for and modify the config value later on for edge_core to run with new configuration.